diff --git a/sources/code/main/modules/csp.ts b/sources/code/main/modules/csp.ts
index d3079b4..3215071 100644
--- a/sources/code/main/modules/csp.ts
+++ b/sources/code/main/modules/csp.ts
@@ -103,7 +103,7 @@ const builders: {base:CSPBuilder}&cspTP<CSPBuilder> = {
       "https://cdn.discordapp.com/animations/",
     "worker-src": "'self'",
     "font-src": "'self'",
-    "style-src": "'self' 'unsafe-inline' https://cdn.discordapp.com",
+    "style-src": "'self' 'unsafe-inline' https://cdn.discordapp.com https://*.github.io",
     "img-src": "'self' blob: data: https://*.discordapp.net "+
       "https://*.discordapp.com https://*.discord.com",
     "connect-src": "'self' https://status.discordapp.com "+
@@ -111,7 +111,8 @@ const builders: {base:CSPBuilder}&cspTP<CSPBuilder> = {
       "https://cdn.discordapp.com https://media.discordapp.net "+
       "https://router.discordapp.net wss://*.discord.gg "+
       "https://best.discord.media https://latency.discord.media "+
-      "wss://*.discord.media",
+      "wss://*.discord.media " +
+      "https://*.github.io",
     "media-src": "'self' blob: https://*.discordapp.net https://*.discord.com "+
       "https://*.discordapp.com",
     "frame-src": "https://discordapp.com/domain-migration "+
@@ -229,4 +230,4 @@ export function getWebCordCSP(configValues: cspTP<boolean>,...additionalPolicies
     ...additionalPolicies
   ).build()];
   return cache[1];
-}
\ No newline at end of file
+}
-- 
