--- neofetch	2022-11-20 11:01:19.790687392 +0300
+++ nicksfetch	2022-11-20 12:21:56.635943605 +0300
@@ -999,6 +999,13 @@
                     *) distro="OS Elbrus $(< /etc/mcst_version)"
                 esac
 
+            elif [[ -f /etc/NIXOS ]]; then
+                case $distro_shorthand in
+                    on) distro="Nick's OS $(nixos-version | awk '{print substr($1,0,5),$2}')" ;;
+                    tiny) distro="Nick's OS" ;;
+                    *) distro="Nick's OS $(nixos-version)" ;;
+                esac
+
             elif type -p pveversion >/dev/null; then
                 case $distro_shorthand in
                     on|tiny) distro="Proxmox VE" ;;
@@ -8973,7 +8980,7 @@
 EOF
         ;;
 
-        "Nicks OS"*)
+        "Nick's OS"*)
             set_colors 4 6
             read -rd '' ascii_data <<'EOF'
 ${c1}          ▗▄▄▄       ${c2}▗▄▄▄▄    ▄▄▄▖
