--- neofetch	2022-11-20 06:03:20.894608138 +0300
+++ neofetch-patched	2022-11-20 06:09:02.740072462 +0300
@@ -5155,7 +5155,7 @@
                                 LaxerOS, LibreELEC, LFS, Linux_Lite, LMDE, Lubuntu, Lunar, macos,
                                 Mageia, MagpieOS, Mandriva, Manjaro, TeArch, Maui, Mer, Minix, LinuxMint,
                                 Live_Raizo, MX_Linux, Namib, Neptune, NetBSD, Netrunner, Nitrux,
-                                NixOS, Nurunner, NuTyX, OBRevenge, OpenBSD, openEuler, OpenIndiana,
+                                NicksOS, Nurunner, NuTyX, OBRevenge, OpenBSD, openEuler, OpenIndiana,
                                 openmamba, OpenMandriva, OpenStage, OpenWrt, osmc, Oracle,
                                 OS Elbrus, PacBSD, Parabola, Pardus, Parrot, Parsix, TrueOS,
                                 PCLinuxOS, Pengwin, Peppermint, Pisi, popos, Porteus, PostMarketOS,
@@ -8973,7 +8973,7 @@
 EOF
         ;;
 
-        "NixOS"*)
+        "NicksOS"*)
             set_colors 4 6
             read -rd '' ascii_data <<'EOF'
 ${c1}          ▗▄▄▄       ${c2}▗▄▄▄▄    ▄▄▄▖
