--- neofetch	2022-11-20 11:01:19.790687392 +0300
+++ nicksfetch	2022-11-20 11:10:10.949638551 +0300
@@ -8935,7 +8935,7 @@
 EOF
         ;;
 
-        "nixos_small")
+        "Nick's OS_small")
             set_colors 4 6
             read -rd '' ascii_data <<'EOF'
   ${c1}  \\\\  \\\\ //
@@ -8948,7 +8948,7 @@
 EOF
         ;;
 
-        "nixos_old"*)
+        "Nick's OS_old"*)
             set_colors 4 6
             read -rd '' ascii_data <<'EOF'
 ${c1}          ::::.    ${c2}':::::     ::::'
@@ -8973,7 +8973,7 @@
 EOF
         ;;
 
-        "Nicks OS"*)
+        "Nick's OS"*)
             set_colors 4 6
             read -rd '' ascii_data <<'EOF'
 ${c1}          ▗▄▄▄       ${c2}▗▄▄▄▄    ▄▄▄▖
